### DISCUSSION TITLE
Lip-sync'ed animated-head avatar in gallery **UPDATED**

### DISCUSSION THREAD

**Discussion by sarutobiumon** [oobabooga/text-generation-webui#1449] [Ideas]
- Would be amazing if you can make an extension for the use case below. 
1. Turn the gallery persona images into animated gifs lip-sync'ed to the wav audio generated by tts (using Bark oobabooga extention below or the latest Coqui TTS which is massively improved) 
2. Then generating an mp4 from the combination of animated gif and wav audio on the fly, replacing the starting-point animated gif on the screen. 

This can be done by integrating code from one of the following choices:
- BEST option:  using Sadtalker https://github.com/Winfredy/SadTalker
 Demos:
1: https://www.youtube.com/watch?v=vzleOzohKNA
2: https://www.youtube.com/watch?v=aJIq_UoZv24
- or this one-click install wav2lip GUI: https://www.youtube.com/watch?v=f_NUZDBiaZg 
- or this google colab python code below (supports 30+ languages): 
https://spltech.co.uk/using-wav2lip-and-google-cloud-wavenet-to-create-voice-overs-in-more-than-30-languages/
- or using VideoReTalking：Audio-based Lip Synchronization for Talking Head Video
https://colab.research.google.com/github/vinthony/video-retalking/blob/main/quick_demo.ipynb
Demo https://www.youtube.com/watch?v=CgZVKSkdtRo

Bark oobabooga tts extention:
https://github.com/wsippel/bark_tts

**Comment by sarutobiumon** [comment]
This would be amazing addition, please look into it since the code is already available.

**Comment by GitHub1712** [comment]
So the ia will be able ta talk with my voice and face? Super cool!

**Reply by sarutobiumon** [reply]
Yeah, exactly! It would bring to life the persona you're talking to via an animated gif, lip-sync'ed to audio generated via TTS with Bark since Bark uses emotions and is the best free equivalent to Eleven Labs, even better actually.

**Comment by sarutobiumon** [comment]
Can the person who made all the pieces work in the video below share the details of how they got these things working together.
This looks like Sadtalker + SableDiffusion + Bark + Oobabooga.
https://www.youtube.com/watch?v=kIaFq7EahqQ

Also, looks like Sadtalker got upgraded to support videos as well!
https://github.com/OpenTalker/SadTalker

**Comment by GitHub1712** [comment]
This needs to be connected to all messengers like teams. To parse the input from teams and give the answer back as webcam + mic.

**Reply by sarutobiumon** [reply]
I think this python code below might actually be the best thing I've seen so far, even better than Sadtalker. It takes both an image or a video as input and generates near-realistic outputs: 

https://colab.research.google.com/github/justinjohn0306/Wav2Lip/blob/master/Wav2Lip_simplified_v5.ipynb#scrollTo=U1xFNFU58_2j

Demo: https://www.youtube.com/watch?v=FkfEvzUlVJk
@oobabooga this would be amazing to integrate into the gallery on your awesome webui. Maybe as a long term to-do extension :)

**Comment by sarutobiumon** [comment]
Latest Coqui TTS released a few hours ago, massively improved, FYI.
